sdk/eventhub/azure-eventhub/tests/livetest/synctests/test_reconnect.py

```diff
@@ -164,8 +164,7 @@ def on_event_received(event):
                 consumer._handler.do_work()
                 assert consumer._handler._connection._state == uamqp.c_uamqp.ConnectionState.DISCARDING
             else:
-                with pytest.raises(error.AMQPConnectionError):
```

Comment:
Has this changed because `keep-alive` got added back?

Reply:
yes, it no longer raises the error due to the `keep-alive` being active.

Reply:
update - this test was changed due to an update made to `_client_run` in `ReceiveClient`. Previously calling `do_work` would always issue a flow and a bug fix went in to make sure that happened only when link credits are 0. 

This changed the test as the flow was no longer being issued on a down connection, leading to an error.  The behavior matches uamqp now. I think the link credit check should be pushed down to the link layer, similar to uamqp